### PR TITLE
Support for nova-drain

### PR DESCRIPTION
**Pull request by DTadrzak**
Signed-off-by: DTadrzak <daniel.tadrzak@intel.com>

### CODE REVIEW

File: nova/templates/configmaps/drain-conf.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.yaml: |+
+    IdentityEndpoint: {{ keystone.auth_uri }}/v3/
+    Username: {{ keystone.admin_user }}
+    Password: {{ keystone.admin_password }}
+    TenantName: {{ drain.tenant_name }}
+    DomainID: {{ drain.domain_id }}
```
Comment (Quentin-M) [inline nova/templates/configmaps/drain-conf.yaml.j2]: ^

**Comment by DTadrzak** [comment]
@Quentin-M 
@mateusz-blaszkowski 
@ss7pro 
@PiotrProkop

File: nova/manifest.jsonnet [hunk 2]
```diff
@@ -65,6 +66,10 @@ kpm.package({
       },
     },
 
+    drain: {
```
Comment (Quentin-M) [inline nova/manifest.jsonnet]: Done

File: nova/manifest.jsonnet [hunk 3]
```diff
@@ -136,6 +141,13 @@ kpm.package({
     },
 
     {
+      file: "configmaps/drain-conf.yaml.j2",
+      template: (importstr "templates/configmaps/drain-conf.yaml.j2"),
+      name: "drain-conf",
```
Comment (Quentin-M) [inline nova/manifest.jsonnet]: Done.

File: nova/templates/compute/daemonset.yaml.j2 [hunk 4]
```diff
@@ -118,7 +118,43 @@ spec:
...
+              value: "/config.yaml,/etc/resolv.conf"
+          volumeMounts:
+            - name: drainconf
+              mountPath: /configmaps/config.yaml
+            - name: resolvconf
+              mountPath: /configmaps/resolv.conf
       volumes:
+        - name: drainconf
+          configMap:
+            name: drain-conf
```
Comment (DTadrzak) [inline nova/templates/compute/daemonset.yaml.j2]: Thanks!

File: nova/templates/compute/drain.yaml.j2 [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        app: nova-drain
+      {% if deployment.engine == "rkt" -%}
+      annotations:
+        rkt.alpha.kubernetes.io/stage1-name-override: coreos.com/rkt/stage1-fly
+      {%- endif %}
+    spec:
+      nodeSelector:
+        {{ deployment.compute_node_label }}: enabled
+      securityContext:
+        runAsUser: 0
```
Comment (mateusz-blaszkowski) [inline nova/templates/compute/drain.yaml.j2]: Can we remove it?

**Comment by PiotrProkop** [comment]
Add dependencies for nova-compute and nova-libvirt.

**Comment by PiotrProkop** [comment]
LGTM.

**Pull request merged**